scripts/check_backend_associated_test_file_test.py

```diff
@@ -45,9 +45,10 @@ def mock_error(msg: str) -> None:
 
     def test_checks_fail_when_a_backend_file_lacks_associated_test_file(
             self) -> None:
-        tempdir = tempfile.TemporaryDirectory(prefix=os.getcwd() + '/core/')
-        backend_file = os.path.join(tempdir.name, 'backend_file.py')
-        frontend_file = os.path.join(tempdir.name, 'frontend_file.ts')
+        with tempfile.TemporaryDirectory(
+            prefix=os.getcwd() + '/core/') as tempdir:
+            backend_file = os.path.join(tempdir, 'backend_file.py')
```

Comment:
Hey @m0nax5, for the backend test, I think here it should be `tempdir.name`, could you please check if after this the backend tests passes, same changes goes for the other backend test below as well.
Thanks

Reply:
Hi @lkbhitesh07, I implemented the changes that you suggested accross the file and the backend tests still won't pass. Here is the log I got for the backend tests :

```
======================================================================
ERROR: test_checks_fail_when_a_backend_file_lacks_associated_test_file (scripts.check_backend_associated_test_file_test.CheckBackendAssociatedTestFileTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/opensource2/oppia/scripts/check_backend_associated_test_file_test.py", line 54, in test_checks_fail_when_a_backend_file_lacks_associated_test_file
    with open(backend_file, 'w', encoding='utf8') as f:
FileNotFoundError: [Errno 2] No such file or directory: '/home/opensource2/oppia/core/z0twcbrp/backend_file.py'

======================================================================
ERROR: test_checks_pass_when_all_backend_files_have_an_associated_test_file (scripts.check_backend_associated_test_file_test.CheckBackendAssociatedTestFileTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/opensource2/oppia/scripts/check_backend_associated_test_file_test.py", line 97, in test_checks_pass_when_all_backend_files_have_an_associated_test_file
    backend_file = os.path.join(tempdir.name, 'backend_file.py')
AttributeError: 'str' object has no attribute 'name'

======================================================================
ERROR: test_pass_when_file_in_exclusion_list_lacks_associated_test (scripts.check_backend_associated_test_file_test.CheckBackendAssociatedTestFileTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/opensource2/oppia/scripts/check_backend_associated_test_file_test.py", line 76, in test_pass_when_file_in_exclusion_list_lacks_associated_test
    backend_file = os.path.join(tempdir.name, 'backend_file.py')
AttributeError: 'str' object has no attribute 'name'

---------------------------------------------------------------------
```
The first error is an occurrence of the previous syntax without `.name` that I left on purpose, whereas the two last errors are what I get when I add `.name`. What I don't understand is why when using the `with` clause tempdir has type `str`, whereas when it is declared with `tempdir = tempfile.TemporaryDirectory()` it has type `tempfile.TemporaryDirectory` and thus `tempdir.name` makes sense and has type `str`. I'm currently looking into finding a solution to this issue.
Thanks again for the review !